diff --git a/fs/fuse/backing.c b/fs/fuse/backing.c
--- a/fs/fuse/backing.c
+++ b/fs/fuse/backing.c
@@ -1429,6 +1429,8 @@ int fuse_mknod_backing(
 	err = vfs_mknod(&nop_mnt_idmap, backing_inode, backing_path.dentry, mode,
 	                new_decode_dev(fmi->rdev));
 	inode_unlock(backing_inode);
+	if (fuse_bpf_lookup_revalidate_enabled && err == -EEXIST)
+		err = -ENOENT;
 	if (err)
 		goto out;
 	if (d_really_is_negative(backing_path.dentry) ||
@@ -1505,6 +1507,8 @@ int fuse_mkdir_backing(
 	if (!IS_POSIXACL(dir_backing_inode))
 		mode &= ~fmi->umask;
 	err = vfs_mkdir(&nop_mnt_idmap, dir_backing_inode, backing_path.dentry, mode);
+	if (fuse_bpf_lookup_revalidate_enabled && err == -EEXIST)
+		err = -ENOENT;
 	if (err)
 		goto out;
 	if (d_really_is_negative(backing_path.dentry) ||
@@ -1864,6 +1868,8 @@ int fuse_link_backing(struct fuse_bpf_args *fa, struct dentry *entry,
 	err = vfs_link(backing_old_path.dentry, &nop_mnt_idmap,
 		       backing_dir_inode, backing_new_path.dentry, NULL);
 	inode_unlock(backing_dir_inode);
+	if (fuse_bpf_lookup_revalidate_enabled && err == -EEXIST)
+		err = -ENOENT;
 	if (err)
 		goto out;
 
@@ -2262,6 +2268,8 @@ int fuse_symlink_backing(
 	inode_lock_nested(backing_inode, I_MUTEX_PARENT);
 	err = vfs_symlink(&nop_mnt_idmap, backing_inode, backing_path.dentry, link);
 	inode_unlock(backing_inode);
+	if (fuse_bpf_lookup_revalidate_enabled && err == -EEXIST)
+		err = -ENOENT;
 	if (err)
 		goto out;
 	if (d_really_is_negative(backing_path.dentry) ||
