diff --git a/fs/fuse/backing.c b/fs/fuse/backing.c
index 2c93713ae3a00..d0fd03afb034c 100644
--- a/fs/fuse/backing.c
+++ b/fs/fuse/backing.c
@@ -1578,6 +1578,8 @@ int fuse_mknod_backing(
 	err = vfs_mknod(&nop_mnt_idmap, backing_inode, backing_path.dentry, mode,
 	                new_decode_dev(fmi->rdev));
 	inode_unlock(backing_inode);
+	if (err == -EEXIST)
+		err = -ENOENT;
 	if (err)
 		goto out;
 	if (d_really_is_negative(backing_path.dentry) ||
@@ -1653,6 +1655,8 @@ int fuse_mkdir_backing(
 	if (!IS_POSIXACL(dir_backing_inode))
 		mode &= ~fmi->umask;
 	err = vfs_mkdir(&nop_mnt_idmap, dir_backing_inode, backing_path.dentry, mode);
+	if (err == -EEXIST)
+		err = -ENOENT;
 	if (err)
 		goto out;
 	if (d_really_is_negative(backing_path.dentry) ||
@@ -2016,6 +2020,8 @@ int fuse_link_backing(struct fuse_bpf_args *fa, struct dentry *entry,
 	err = vfs_link(backing_old_path.dentry, &nop_mnt_idmap,
 		       backing_dir_inode, backing_new_path.dentry, NULL);
 	inode_unlock(backing_dir_inode);
+	if (err == -EEXIST)
+		err = -ENOENT;
 	if (err)
 		goto out;
 
@@ -2409,6 +2415,8 @@ int fuse_symlink_backing(
 	inode_lock_nested(backing_inode, I_MUTEX_PARENT);
 	err = vfs_symlink(&nop_mnt_idmap, backing_inode, backing_path.dentry, link);
 	inode_unlock(backing_inode);
+	if (err == -EEXIST)
+		err = -ENOENT;
 	if (err)
 		goto out;
 	if (d_really_is_negative(backing_path.dentry) ||
